BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/signal/MeanFilter.java

```diff
@@ -0,0 +1,36 @@
+package com.nexenio.bleindoorpositioning.ble.beacon.signal;
+
+import com.nexenio.bleindoorpositioning.ble.advertising.AdvertisingPacket;
+import com.nexenio.bleindoorpositioning.ble.beacon.Beacon;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by leon on 03.01.18.
+ */
+
+public class MeanFilter extends WindowFilter {
+
+    public MeanFilter() {
+    }
+
+    public MeanFilter(long duration, TimeUnit timeUnit) {
+        super(duration, timeUnit);
+    }
+
+    @Override
+    public float filter(Beacon beacon) {
+        float sum = 0;
+        int count = 0;
+        List<AdvertisingPacket> recentAdvertisingPackets = getRecentAdvertisingPackets(beacon);
+        for (AdvertisingPacket advertisingPacket : recentAdvertisingPackets) {
+            if (advertisingPacket.getTimestamp() < minimumTimestamp) {
+                continue;
+            }
+            sum += advertisingPacket.getRssi();
+            count++;
+        }
```

Comment:
Move the mathematical logic to a static method and call it from here. Maybe with a method that get all the `rssi` values and than you calculate the mean for a list of floats.

Reply:
Should already be available in the `AdvertisingPacketUtil`